tests/sentry/sentry_apps/api/endpoints/test_sentry_app_webhook_requests.py

```diff
@@ -0,0 +1,435 @@
+                },
+            )
+
+        url = reverse("sentry-api-0-sentry-app-webhook-requests", args=[self.published_app.slug])
+        errors_only_response = self.client.get(f"{url}?errorsOnly=true", format="json")
+        assert errors_only_response.status_code == 200
+        assert len(errors_only_response.data) == 1
+        assert errors_only_response.data[0] == {
+            "webhookUrl": self.published_app.webhook_url,
+            "sentryAppSlug": self.published_app.slug,
+            "eventType": "issue.assigned",
+            "responseCode": 500,
+            "project_id": 1,
+            "date": str(now) + "+00:00",
+            "error_id": "abc123",
+            "request_body": json.dumps(self.mock_request.body),
+            "request_headers": {"Content-Type": "application/json"},
+            "response_body": json.dumps(self.mock_response.content),
+            "organization": {"name": self.org.name, "id": self.org.id, "slug": self.org.slug},
+        }
+
+        response = self.client.get(url, format="json")
+        assert response.status_code == 200
+        assert len(response.data) == 2
+
+    @with_feature("organizations:sentry-app-webhook-requests")
+    def test_linked_error_not_returned_if_project_does_not_exist(self):
+        self.login_as(user=self.user)
+
+        self.store_event(
+            data={"event_id": self.event_id, "timestamp": before_now(minutes=1).isoformat()},
+            project_id=self.project.id,
+        )
+
+        buffer = SentryAppWebhookRequestsBuffer(self.published_app)
+        buffer.add_request(
+            response_code=200,
+            org_id=self.org.id,
+            event="issue.assigned",
+            url=self.unpublished_app.webhook_url,
+            error_id=self.event_id,
+            project_id=1000,
+        )
+
+        url = reverse("sentry-api-0-sentry-app-webhook-requests", args=[self.published_app.slug])
+        response = self.client.get(url, format="json")
+        assert response.status_code == 200
+        assert len(response.data) == 1
+        assert response.data[0]["organization"]["slug"] == self.org.slug
+        assert response.data[0]["sentryAppSlug"] == self.published_app.slug
+        assert "errorUrl" not in response.data[0]
+
+    @with_feature("organizations:sentry-app-webhook-requests")
+    def test_org_slug_filter(self):
+        """Test that filtering by the qparam organizationSlug properly filters results"""
+        self.login_as(user=self.user)
+        buffer = SentryAppWebhookRequestsBuffer(self.published_app)
+        buffer.add_request(
+            response_code=200,
+            org_id=self.org.id,
+            event="issue.assigned",
+            url=self.published_app.webhook_url,
+        )
+        buffer.add_request(
+            response_code=200,
+            org_id=self.org.id,
+            event="issue.assigned",
+            url=self.published_app.webhook_url,
+        )
```

Comment:
For the future, it may be easier to test this type of filtering logic at the util/buffer layer instead of doing it via the API. You'll have strong type guarantees now that we've added typed dicts for responses